File: oj.uz/IOI18_highway/pas/highway.pas
unit highway;

interface

type
  array_of_longint = array of longint;

procedure find_pair(N: longint; U, V: array_of_longint; A, B: longint);

implementation

uses highway_lib;

procedure find_pair(N: longint; U, V: array_of_longint; A, B: longint);
var
  M, i, j: longint;
  w: array of longint;
  toll: int64;
begin
  M := length(U);
  setlength(w, M);
  for j := 0 to 49 do begin
    for i := 0 to M - 1 do begin
      w[i] := 0;
    end;
    toll := ask(w);
  end;
  answer(0, N - 1);
end;

end.

File: oj.uz/IOI18_highway/java/highway.java
public class highway {
  void find_pair(int N, int[] U, int[] V, int A, int B) {
    int M = U.length;
    for (int j = 0; j < 50; ++j) {
      int[] w = new int[M];
      for (int i = 0; i < M; ++i) {
        w[i] = 0;
      }
      long toll = highway_lib.ask(w);
    }
    highway_lib.answer(0, N - 1);
  }
}
